metrics/perplexity/test_perplexity.py

```diff
@@ -0,0 +1,36 @@
+import datasets
+from datasets import load_dataset
+
+
+def main():
+    print("\n\n\n------------  WIKITEXT:  ------------")
+    input_texts = load_dataset("wikitext", "wikitext-2-raw-v1", keep_in_memory=False, split="test")["text"][:50]
+    print("input len with empty strings:", len(input_texts))
+    input_texts = [s for s in input_texts if s != ""]
+    print("input len without empty strings:", len(input_texts))
+
+    perplexity = datasets.load_metric("../perplexity")
+
+    results_w_start = perplexity.compute(input_texts=input_texts, model_id="gpt2", device="gpu", add_start_token=True)
+    print("\n\nRES W START:", results_w_start)
+
+    results_no_start = perplexity.compute(
+        input_texts=input_texts, model_id="gpt2", device="gpu", add_start_token=False
+    )
+    print("\n\nRES NO START:", results_no_start)
+
+    print("\n\n\n------------  SMALL SNIPPETS:  ------------")
+    input_texts = ["lorem ipsum", "Happy Birthday!", "Bienvenue"]
+    print(input_texts)
+
+    results_w_start = perplexity.compute(input_texts=input_texts, model_id="gpt2", device="gpu", add_start_token=True)
+    print("\n\nRES W START:", results_w_start)
+
+    results_no_start = perplexity.compute(
+        input_texts=input_texts, model_id="gpt2", device="gpu", add_start_token=False
+    )
+    print("\n\nRES NO START:", results_no_start)
```

Comment:
For reference, the results I get from this code are below. Note that they are all extremely high, and many are `inf`:

`--- WIKITEXT: ---
RES W START: {'perplexity': [5.26044112490735e+33, 1754766870839296.0, 4.851017998674971e+21, 1.6137719812847106e+33, 1.123557053605896e+35, 1.4592395342302413e+17, 2.2242306063538344e+21, 1.6256261294134198e+37, 5.6552604843696494e+23, 5.283058871869749e+23, 1.362616559062237e+31, 3.2894723458301157e+30, inf, 2.8323108695319055e+32, 4.517404432663069e+36, 6.06021648422544e+21, 2.7673238221998803e+25, inf, 1.4324680994914391e+32, 1.0707699658437287e+29, 1.3665097609664754e+23, 2.646539983794732e+26, 1.1760327816258613e+21, 5.623264614959543e+28, 8.202277198380542e+22, 1.6075579260687217e+30, 7.034012837768928e+25]}

RES NO START: {'perplexity': [6.347511649255629e+27, 7.746963798754648e+35, 5.412155643337211e+36, 7.443727567431987e+26, 7.610168923774734e+33, 6.160369930664121e+35, 5.744005654242058e+36, 6.396653209350697e+32, inf, inf, 4.597405287035285e+27, 1.5910406783685568e+27, 1.2694701550401124e+26, 9.076024979217206e+25, 2.580195144338176e+31, inf, 1.8766434173654982e+35, 1.4048184593730499e+29, inf, 5.501222136759501e+28, inf, inf, 1.6581890056188304e+38, inf, inf, inf, inf]}


--- SMALL SNIPPETS: ---
RES W START: {'perplexity': [2.6583647714047857e+31, 4.2388171843120376e+26, 3.074649231109679e+28]}

RES NO START: {'perplexity': [2.605767698770934e+26, 7.023913198776316e+20, 1.3889064307219916e+20]}`